src/nvim/ex_cmds2.c

```diff
@@ -259,7 +292,28 @@ void do_debug(char_u *cmd)
           /* Do not repeat ">interrupt" cmd, continue stepping. */
           last_cmd = CMD_STEP;
           break;
+        case CMD_BACKTRACE:
+          do_showbacktrace(cmd);
+          continue;
+        case CMD_FRAME:
+          if (*p == NUL) {
+            do_showbacktrace(cmd);
+          } else {
+            p = skipwhite(p);
+            do_setdebugtracelevel(p);
+          }
+          continue;
+        case CMD_UP:
+          debug_backtrace_level++;
+          do_checkbacktracelevel();
+          continue;
+        case CMD_DOWN:
+          debug_backtrace_level--;
+          do_checkbacktracelevel();
+          continue;
```

Comment:
The introduction of `continue` here and above means `xfree(cmdline);` is never hit.

If you fix this memory leak please make it a separate commit. cc @chrisbra